server/src/main/java/io/unitycatalog/server/service/Scim2UserService.java

```diff
@@ -0,0 +1,204 @@
+import io.unitycatalog.server.exception.ErrorCode;
+import io.unitycatalog.server.exception.GlobalExceptionHandler;
+import io.unitycatalog.server.exception.Scim2RuntimeException;
+import io.unitycatalog.server.persist.UserRepository;
+import java.net.URI;
+import java.util.Calendar;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * SCIM2-compliant user management.
+ *
+ * <p>This will be a SCIM 2.0 compliant user management service. The UC User data model will be a
+ * minimal set of required fields that are necessary to support user management/exchange.
+ *
+ * <ul>
+ *   <li>id - internal unique identifier for user
+ *   <li>name - maps to SCIM displayName
+ *   <li>email - maps to SCIM primary email
+ *   <li>externalId - maps to SCIM external id
+ *   <li>userName - maps to SCIM primary email
+ * </ul>
+ */
+@ExceptionHandler(GlobalExceptionHandler.class)
+public class Scim2UserService {
+  private static final UserRepository USER_REPOSITORY = UserRepository.getInstance();
+
+  public Scim2UserService() {}
+
+  @Get("")
+  public HttpResponse getScimUsers(
+      @Param("filter") Optional<String> filter,
+      @Param("startIndex") Optional<Integer> startIndex,
+      @Param("count") Optional<Integer> count) {
+    final Filter userFilter = filter.filter(f -> !f.isEmpty()).map(this::parseFilter).orElse(null);
+    FilterEvaluator filterEvaluator = new FilterEvaluator();
+
+    List<UserResource> userResourcesList =
+        USER_REPOSITORY.listUsers().stream()
+            .map(this::asUserResource)
+            .filter(m -> match(filterEvaluator, userFilter, m))
+            .skip(startIndex.orElse(1) - 1)
+            .limit(count.orElse(50))
+            .toList();
+    return HttpResponse.ofJson(userResourcesList);
+  }
+
+  @Post("")
+  public HttpResponse createScimUser(UserResource userResource) {
+    // Get primary email address
+    Email primaryEmail =
+        userResource.getEmails().stream()
+            .filter(Email::getPrimary)
+            .findFirst()
+            .orElseThrow(
+                () ->
+                    new Scim2RuntimeException(
+                        new PreconditionFailedException("User does not have a primary email.")));
+
+    try {
+      User user = USER_REPOSITORY.getUserByEmail(primaryEmail.getValue());
+      return HttpResponse.ofJson(asUserResource(user));
+    } catch (BaseException e) {
+      if (e.getErrorCode() == ErrorCode.NOT_FOUND) {
+        String pictureUrl = "";
+        if (userResource.getPhotos() != null && !userResource.getPhotos().isEmpty()) {
+          pictureUrl = userResource.getPhotos().get(0).getValue().toString();
+        }
+        User user =
+            USER_REPOSITORY.createUser(
+                new CreateUser()
+                    .name(userResource.getDisplayName())
+                    .email(primaryEmail.getValue())
+                    .externalId(userResource.getExternalId())
+                    .pictureUrl(pictureUrl));
```

Comment:
Are the `meta` and `active` fields of the `UserResource` that could be sent as part of a create user request ignored?

Reply:
we should process `active`, even though that would be strange to create a disabled user, but I suppose a possibility.  `meta` we do ignore.

Reply:
Done.